The --export-compile-commands option for build-webkit is broken.

shivamidow opened the pull request:
#### 0579d0e2c508a74054d935809ff429b4caad7ee2
<pre>
The --export-compile-commands option for build-webkit is broken.
<a href="https://bugs.webkit.org/show_bug.cgi?id=264239">https://bugs.webkit.org/show_bug.cgi?id=264239</a>

Reviewed by Elliott Williams.

The build-webkit with --export-compile-commands on macOS does not produce
compile_commands.json. In bug 235686 [1], we added the support of exporting
compile_commands.json to the make build. This change brings the same
support into the build-webkit script.

[1] <a href="https://webkit.org/b/235686">https://webkit.org/b/235686</a>

* Tools/Scripts/build-webkit:
* Tools/Scripts/webkitdirs.pm:
(XcodeExportCompileCommandsOptions):
* Tools/Scripts/webkitperl/BuildSubproject.pm:

Canonical link: <a href="https://commits.webkit.org/270348@main">https://commits.webkit.org/270348@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0442f7ab40f95e69be293adb5e909061cac85edd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25000 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3543 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26253 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27116 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22951 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25268 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5240 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/982 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23227 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25244 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2577 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21576 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27696 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2273 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22512 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28639 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/21740 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22807 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22867 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26464 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/24250 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2220 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/520 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/31656 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3501 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6933 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6050 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2665 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/31641 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2563 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/6612 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
Thank you! This seems reasonable to me, and I assume you're already using it locally :)

For future refactoring, I would like us to pass the args from `XcodeOptions()` in webkitdirs. That subroutine is used by Make and `build-webkit`, so if we pass it there we can share the list of hard-coded build setting overrides everywhere, and change `Makefile.shared` to just pass `--export-compile-commands` instead of computing its own build setting arguments.

shivamidow committed:
The --export-compile-commands option for build-webkit is broken.
https://bugs.webkit.org/show_bug.cgi?id=264239

Reviewed by Elliott Williams.

The build-webkit with --export-compile-commands on macOS does not produce
compile_commands.json. In bug 235686 [1], we added the support of exporting
compile_commands.json to the make build. This change brings the same
support into the build-webkit script.

[1] https://webkit.org/b/235686

* Tools/Scripts/build-webkit:
* Tools/Scripts/webkitdirs.pm:
(XcodeExportCompileCommandsOptions):
* Tools/Scripts/webkitperl/BuildSubproject.pm:

Canonical link: https://commits.webkit.org/270348@main
